lib/statement.rb

```diff
@@ -0,0 +1,43 @@
+class Statement
+  attr_reader :ledger, :formatted_transactions, :balance
+
+  def initialize(ledger)
+    @ledger = ledger
+    @formatted_transactions = []
+    @balance ||= 0
+  end
+
+  def print_statement
+    format_transactions
+    print_heading
+    print_transactions
+  end
+
+  private
+
+  def print_heading
+    puts 'date || credit || debit || balance'
+  end
+
+  def format_transactions
+    @ledger.transactions.each do |record|
+      @formatted_transactions.push(amount: record.amount, date: record.date, type: record.type)
```

Comment:
Do these transactions need to be formatted before being printed? It feels like the main thing the formatting does is to put each transaction into a hash.